### PR TITLE
feat: 🤖 add new tree shaking

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**

### CODE REVIEW

File: docs/en/config/experiments.mdx [hunk 1]
```diff
@@ -265,3 +265,22 @@ module.exports = {
...
+### experiments.rspackFuture.newTreeshaking
+
+<ApiMeta addedVersion="0.4.2" />
+
+- **Type:** `boolean`
+- **Default Value:** `false`
+
+This feature enables the new treeshaking implementation the same as webpack, which would generate more efficient and smaller code
+
+```js title="rspack.config.js"
```
Comment (hardfist) [inline docs/en/config/experiments.mdx:278]: should we disable builtins.treeShaking at same time?

Comment (hardfist) [reply docs/en/config/experiments.mdx:278]: what happens if we disable bultins.treeShaking and enable rspackFuture.newTreeshaking at same time?

File: docs/en/config/optimization.mdx [hunk 2]
```diff
@@ -99,6 +99,12 @@ module.exports = {
 
 Remove empty chunks generated in the compilation.
 
+## optimization.removeEmptyChunks
```
Comment (hardfist) [inline docs/en/config/optimization.mdx:102]: this is enabled in development too?

File: docs/en/config/experiments.mdx (hunk 1 above)
Comment (IWANABETHATGUY) [reply docs/en/config/experiments.mdx:278]: I will open a pull request to do that.

**Review by hardfist — APPROVED** [review]

**Pull request merged**